src/gnutls.c

```diff
@@ -1021,7 +1021,8 @@ static int copycat (int local, int remote, gnutls_session_t wrapped, pool_handle
 	struct linger linger = { 1, 10 };
 	int have_client;
 	int retval = GNUTLS_E_SUCCESS;
-
+	int local_shutdown_done = 0;
+	int remote_shutdown_done = 0;
 client = INVALID_POOL_HANDLE;
```

Comment:
1. deze flags kun je ook laten terugkeren in de while op regel 1042/1043 lijkt me?  beter dan POLLIN dat een momentopname is waarna we eventueel nog door willen lezen (maar niet meer schrijven)

2. w/s stijl: mag er wel weer een lege regel na de declaraties?

3. zie onder: negatie van deze flags lijkt mij leesbaarder

4. bool ipv int?

Reply:
#include <stdbool.h>
bool x = true / false;